src/main/java/htsjdk/samtools/CRAMIterator.java

```diff
@@ -209,10 +205,9 @@ void nextContainer() throws IllegalArgumentException, CRAMException {
             if (mReader != null) {
                 final long chunkStart = (container.offset << 16) | cramRecord.sliceIndex;
                 final long chunkEnd = ((container.offset << 16) | cramRecord.sliceIndex) + 1;
-                nextRecord.setFileSource(new SAMFileSource(mReader,
-                        new BAMFileSpan(new Chunk(chunkStart, chunkEnd))));
+                samRecord.setFileSource(new SAMFileSource(mReader, new BAMFileSpan(new Chunk(chunkStart, chunkEnd))));
```

Comment:
This appears to be what was intended here.

Reply:
Apparently this code block has never been executed before, since it would have gotten an NPE. Can you add a test that exercises this (calls `enableFileSource` on a CRAMReader, and verifies that `SAMRecord.getFileSource` returns a `SAMFileSource` in that case) ?

Reply:
Writing this test was a bit tricky because I found that this path is only triggered when `cramFile` is null - i.e. the `CRAMIterator` was constructed using an `InputStream` instead of a file.  Maybe it's the rarity of that situation which hid this bug for so long.